Src/Sarif.PatternMatcher.Sdk/ValidatorBase.cs

```diff
@@ -118,15 +118,92 @@ public static bool IsBase64EncodingOfPrintableCharacters(string secret)
             return true;
         }
 
-        public static bool ContainsDigitAndChar(string matchedPattern)
+        public static bool ContainsDigitAndChar(string secret)
         {
-            bool oneDigit = false, oneLetter = false;
+            return ContainAtLeastNDigit(secret, 1) && ContainAtLeastNLetter(secret, 1);
+        }
+
+        public static bool ContainLowerAndUpperCaseLetter(string secret)
+        {
+            return ContainAtLeastNLowerCaseLetter(secret, 1) && ContainAtLeastNUpperCaseLetter(secret, 1);
+        }
+
+        public static bool ContainAtLeastNDigit(string secret, int n)
```

Comment:
@LingZhou-gh Should be called `ContainsAtLeastNDigits`

Reply:
changed